force-app/main/default/lwc/recordPickerDynamicTarget/recordPickerDynamicTarget.html

```diff
@@ -0,0 +1,41 @@
+<template>
+    <lightning-card
+        title="RecordPickerDynamicTarget"
+        class="slds-card__body_inner"
+        icon-name="standard:search"
+    >
+        <div class="slds-form-element">
+            <label class="slds-form-element__label">{label}</label>
+            <div class="slds-form-element__control slds-combobox-group">
+                <template if:true={showTargetSelector}>
+                    <lightning-combobox
+                        data-id="targetSelector"
+                        label="Select a target sObject"
+                        variant="label-hidden"
+                        options={targetObjects}
+                        value={currentObjectApiName}
+                        onchange={handleTargetSelection}
+                        class="slds-combobox_object-switcher slds-combobox-addon_start"
+                    >
+                    </lightning-combobox>
+                </template>
+                <lightning-record-picker
+                    lwc:ref="recordPicker"
+                    object-api-name={currentObjectApiName}
+                    placeholder={placeholder}
+                    label={label}
+                    variant="label-hidden"
+                    display-info={displayInfo}
+                    matching-info={matchingInfo}
+                    onchange={handleRecordSelect}
+                    class="slds-size_full slds-combobox-addon_end"
+                >
+                </lightning-record-picker>
+            </div>
+        </div>
+
+        <c-view-source source="lwc/recordPickerDynamicTarget" slot="footer">
+            Select a target sObject. Type a search term and select a record.
```

Comment:
question/suggestion: I think "object" is the term externally used to design what's internally called an "entity" (Account, Custom__c, ...).
I think that "sObject" is more an Apex term.